### ISSUE TITLE
Synthesis images with GPU use.

### ISSUE THREAD

**Issue by ngonthier** [omrysendik/DCor#4]
Hi,

I try to synthesis HD images (1024*1024) with your Deep Corr code but it is really slow. Moreover it seems that my GPU is not really used by MatConvNet during the iterations. 

Do you know if I have to add some specific option to use the GPU or how I can speed up the algorithm ?

**Comment by omrysendik** [comment]
Sure. You should compile MatConvNet Mex files for GPU.
But, most of the time is spent on the LBFGSB, which didn't have GPU
acceleration when I released my code.
There now exists such a package and you can try and replace mine with it.

**Comment by ngonthier** [comment]
Yes, I already have compiled MatConvNet Mex files for GPU before writing the first message. 
I think you are totally right, most of the time is spent on the LBFGS-B (not on GPU). 
I didn't know that LBFGSB could be implemented to run faster on GPU. Do you know where I can find such package please ? I didn't find it on Mathworks.com.
Thanks.

**Comment by omrysendik** [comment]
I was looking into this in the past but never completed an integration:

https://github.com/nepluno/lbfgsb-gpu

**Comment by ngonthier** [comment]
I will see if I can integrate it. Thanks.

**Comment by ngonthier** [comment]
Hi again,

It seems that the code don't use the GPU just by compiling the Mex files with the GPU option. 
Moreover when I profile the code, LBFGS is not the bottleneck at all, it seems to be the correlation function.

I try to add some gpuArray in the code and to move the net from the CPU to the GPU but I got a lot of  different error most of the time, it seems to be due to the lack of memory on the GPU card. 

Do you already have a version of the code using gpuArray ? Do you know how I could speed up the code ? 

Best,

**Comment by omrysendik** [comment]
Adding gpuArray will not help you because the calculation of the correlation doesn't use the MatConvNet GPU backend. I didn't implement such a function because speed wasn't my emphasis.
